frontend/integration-tests/tests/kubevirt/vm.actions.scenario.ts

```diff
@@ -0,0 +1,162 @@
+
+describe('Test VM actions', () => {
+  const leakedResources = new Set<string>();
+  const testVM = {
+    apiVersion: 'kubevirt.io/v1alpha2',
+    kind: 'VirtualMachine',
+    metadata: {
+      name: testName,
+      namespace: testName,
+    },
+    spec: {
+      running: false,
+      template: {
+        spec: {
+          domain: {
+            cpu: {
+              cores: 1,
+            },
+            devices: {
+              disks: [
+                {
+                  bootOrder: 1,
+                  disk: {
+                    bus: 'virtio',
+                  },
+                  name: 'rootdisk',
+                  volumeName: 'rootdisk',
+                },
+              ],
+              interfaces: [
+                {
+                  bridge: {},
+                  name: 'eth0',
+                },
+              ],
+            },
+            resources: {
+              requests: {
+                memory: '2G',
+              },
+            },
+          },
+          networks: [
+            {
+              name: 'eth0',
+              pod: {},
+            },
+          ],
+          volumes: [
+            {
+              containerDisk: {
+                image: 'kubevirt/cirros-registry-disk-demo:latest',
+              },
+              name: 'rootdisk',
+            },
+          ],
+        },
+      },
+    },
+  };
+
+  afterAll(async() => {
+    const leakedArray: Array<string> = [...leakedResources];
+    if (leakedArray.length > 0) {
+      console.error(`Leaked ${leakedArray.join()}`);
+      leakedArray.map(r => JSON.parse(r) as {name: string, namespace: string, kind: string})
+        .forEach(({name, namespace, kind}) => {
+          try {
+            execSync(`kubectl delete -n ${namespace} --cascade ${kind} ${name}`);
+          } catch (error) {
+            console.error(`Failed to delete ${kind} ${name}:\n${error}`);
+          }
+        });
+    }
+  });
+
+  describe('Test VM list view kebab actions', () => {
+    const vmName = `vm-list-view-actions-${testName}`;
+    beforeAll(async() => {
+      testVM.metadata.name = vmName;
+      execSync(`echo '${JSON.stringify(testVM)}' | kubectl create -f -`);
+      leakedResources.add(JSON.stringify({name: vmName, namespace: testName, kind: 'vm'}));
+    });
+
+    it('Navigates to VMs', async() => {
+      await browser.get(`${appHost}/k8s/all-namespaces/virtualmachines`);
+      await isLoaded();
+      await filterForName(vmName);
+      await resourceRowsPresent();
+    });
+
+    it('Starts VM', async() => {
+      await vmActionsView.listViewAction(vmName)('Start');
+      await browser.wait(until.textToBePresentInElement(vmActionsView.listViewVMmStatus(vmName), 'Running'), 20000);
+    });
+
+    it('Restarts VM', async() => {
+      await isLoaded();
+      await vmActionsView.listViewAction(vmName)('Restart');
+      await browser.wait(until.textToBePresentInElement(vmActionsView.listViewVMmStatus(vmName), 'Running'), 20000);
```

Comment:
Can we test for a change, i.e. `Running -> Starting -> Running`? Not sure if it could be too quick on some environments though?

Reply:
I'm not sure I understand. You can't start already running VM, is it what you mean?

Reply:
Just to be sure the VM restarted. VM already has the status `Running` when clicked on the `Restart` if I am not mistaken. I am just worried we might not catch the case when restart action fails.

Reply:
Ah yes, of course... You're right.

Reply:
Done